### PR TITLE
[physfs] Patches to fix behavior on macOS / Linux

### PR DESCRIPTION
**Pull request by past-due**
**Describe the pull request**

Adds two patches to fix behavior (and runtime failure handling) on macOS / Linux / other POSIX-compatible.

- Handle `EINTR` (fixes runtime failures - more common on macOS Big Sur, and depending on system configuration and many other factors)
- Adds missing `O_CLOEXEC` / `FD_CLOEXEC` (recommended behavior for libraries)

All triplets should be supported.

I have reached out to upstream about these patches. Will update when there's a response.

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]
Can you report this to the upstream?

Thanks.

**Comment by past-due** [comment]
> Can you report this to the upstream?
> 
> Thanks.

@JackBoosY:
I have reported this to the maintainer (no public link though, as unfortunately there doesn't appear to be a public bug-tracker). Will post a comment here with updates whenever I hear back.

**Comment by ras0219-msft** [comment]
I'm uneasy about adding these code changes without some guidance from upstream -- I'm not familiar enough with the specific semantics and codebase to properly review it.

Assuming upstream gives the OK, then this LGTM

**Comment by JackBoosY** [comment]
Ping @past-due for response, does the upstream accept this changes?

**Comment by past-due** [comment]
> Ping @past-due for response, does the upstream accept this changes?

Haven't heard back - I will ping them again.

**Comment by past-due** [comment]
@JackBoosY I'm pleased to note that upstream has now accepted these patches:

https://github.com/icculus/physfs/commit/a9cb20772baa3f042185bd74ab715da033a75e0c
https://github.com/icculus/physfs/commit/b8fa8fdcac75d055059ac2f6d31292e419c8c98c

(They also recently moved to GitHub, which made this a bit easier.)

File: ports/physfs/portfile.cmake [hunk 1]
```diff
@@ -12,7 +12,9 @@ vcpkg_extract_source_archive_ex(
     ARCHIVE ${ARCHIVE}
     REF ${PHYSFS_VERSION}
     PATCHES
-        "fix-lzmasdk-arm64-windows.patch"
+        "001-fix-lzmasdk-arm64-windows.patch"
+        "002-fix-posix-eintr.patch"
+        "003-fix-posix-cloexec.patch"
```
Comment (JackBoosY) [inline ports/physfs/portfile.cmake:17]:
```suggestion
        "003-fix-posix-cloexec.patch" # Remove this patch in the next update
```

File: ports/physfs/portfile.cmake [hunk 2]
```diff
@@ -12,7 +12,9 @@ vcpkg_extract_source_archive_ex(
     ARCHIVE ${ARCHIVE}
     REF ${PHYSFS_VERSION}
     PATCHES
-        "fix-lzmasdk-arm64-windows.patch"
+        "001-fix-lzmasdk-arm64-windows.patch"
+        "002-fix-posix-eintr.patch"
```
Comment (JackBoosY) [inline ports/physfs/portfile.cmake:16]:
```suggestion
        "002-fix-posix-eintr.patch" # Remove this patch in the next update
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

File: versions/p-/physfs.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "66d5438aabcfd4ec4a3e30e0f141fc8e6e5e99ce",
```
Comment (JackBoosY) [inline versions/p-/physfs.json:4]:
```suggestion
      "git-tree": "5976d673e48ec47164efd0ea8ec86728f0e259be",
```

**Pull request merged**